src/main/scala/beam/router/skim/AbstractSkimmer.scala

```diff
@@ -74,8 +76,34 @@ abstract class AbstractSkimmer(beamServices: BeamServices, config: BeamConfig.Be
 
   override def notifyIterationStarts(event: IterationStartsEvent): Unit = {
     if (event.getIteration == 0 && beamConfig.beam.warmStart.enabled) {
-      readOnlySkim.aggregatedSkim =
-        new CsvSkimReader(beamConfig.beam.warmStart.skimsFilePath, fromCsv, logger).readAggregatedSkims
+      val filePath = beamConfig.beam.warmStart.skimsFilePath
+      val file = File(filePath)
+
+      readOnlySkim.aggregatedSkim = if (file.exists) {
+        new CsvSkimReader(filePath, fromCsv, logger).readAggregatedSkims
+      } else {
+        // TODO: skimsFilePath should be list of file. In this case we can read them without scanning directory
+        // TODO: Detecting that huge file or parts is being processed should be moved to BeamWarmStart.scala `skimsFilePath`
```

Comment:
let us make this a separate issue for later

Reply:
https://github.com/LBNL-UCB-STI/beam/issues/2695